GymBuddies/ViewControllers/ProfileViewController.m

```diff
@@ -7,41 +7,214 @@
+                UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+                delegate.window.rootViewController = [storyboard instantiateViewControllerWithIdentifier:@"LoginNavigationController"];
+            }
+        }];
+    }];
+    
+    UIAction *changeProfilePicture = [UIAction actionWithTitle:@"Change Profile Picture" image:nil identifier:nil handler:^(UIAction *action){
+        self.mediaPicker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+        self.mediaPicker.mediaTypes = @[(NSString*)kUTTypeImage];
+        [self presentViewController:self.mediaPicker animated:YES completion:nil];
+    }];
+    
+    UIMenu *menu = [[UIMenu alloc]menuByReplacingChildren:[NSArray arrayWithObjects:logOut, changeProfilePicture, nil]];
+    self.settingsButton.menu = menu;
+    self.settingsButton.showsMenuAsPrimaryAction = YES;
+}
+
+
+#pragma mark - Change Profile Picture
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<UIImagePickerControllerInfoKey,id> *)info{
+    NSURL *urlImage = [info objectForKey:UIImagePickerControllerImageURL];
+    NSString *imageName = urlImage.lastPathComponent;
+    
+    UIImage *editedImage = [info objectForKey:UIImagePickerControllerEditedImage];
+    PFFileObject *imageFile = [ParseAPIManager getPFFileFromImage:editedImage imageName:imageName];
+    
+    [ParseAPIManager changeProfilePicture:imageFile completion:^(BOOL succeeded, NSError * _Nonnull error) {
+        if(succeeded){
+            self.userProfilePicture.image = [info objectForKey:UIImagePickerControllerEditedImage];
         } else {
-            SceneDelegate *delegate = (SceneDelegate *)self.view.window.windowScene.delegate;
-            UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
-            delegate.window.rootViewController = [storyboard instantiateViewControllerWithIdentifier:@"LoginNavigationController"];
+            UIAlertController *alert = [AlertCreator createOkAlert:@"Error changing your profile picture" message:error.localizedDescription];
+            [self presentViewController:alert animated:YES completion:nil];
         }
     }];
+    
+    [self dismissViewControllerAnimated:YES completion:nil];
 }
 
-- (void)viewDidLoad {
-    [super viewDidLoad];
-    // Do any additional setup after loading the view.
+
+#pragma mark - Fetching routines
+
+-(void)fetchUsersCreatedRoutines{
+    [ParseAPIManager fetchUsersCreatedRoutines:^(NSArray * _Nonnull elements, NSError * _Nullable error) {
+            if(elements != nil){
+                self.createdRoutineList = elements;
+                [self.routinesTableView reloadData];
+            } else{
+                UIAlertController *alert = [AlertCreator createOkAlert:@"Error loading your routines" message:error.localizedDescription];
+                [self presentViewController:alert animated:YES completion:nil];
+            }
```

Comment:
refactor lines 128-134 into a separate helper method
this is duplicated also in the method `fetchUsersLikedRoutines`